saq/worker.py

```diff
@@ -219,11 +227,10 @@ async def process(self):
                 else:
                     await job.retry(error)
         finally:
-            if job:
-                self.job_task_contexts.pop(job)
+            if context:
+                self.job_task_contexts.pop(job, None)
 
-                if self.after_process and job.completed:
-                    await self.after_process(context)
+                await self._after_process(context)
```

Comment:
This is a breaking change - 

This always calls `after_process`, even if not completed

Reply:
@vchan will this cause a problem with our current nerf settings that auto commits? @barakalon suggests we rollback instead

Reply:
Yeah, I think it's a best practice to explicitly commit on success. 

Also, I don't know how long nerf jobs are supposed to run, but in general, I'd recommend keeping database transactions _short_. If you begin a transaction, then do a bunch of work not related to that transaction, _then_ finally commit when the job completes, you might cause problems for the database. Especially if that transaction is locking any objects.

Reply:
oooooo didn't realize about the transaction, that's a good thought